src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -1539,34 +1727,44 @@ subroutine initialize_ice_shelf(param_file, ocn_grid, Time, CS, diag, forces, fl
 
   endif ! .not. new_sim
 
+!  do j=G%jsc,G%jec ; do i=G%isc,G%iec
+!    ISS%area_shelf_h(i,j) = ISS%area_shelf_h(i,j)*G%mask2dT(i,j)
+!  enddo; enddo
+
   CS%Time = Time
 
-  call cpu_clock_begin(id_clock_pass)
+  CS%id_clock_shelf = cpu_clock_id('Ice shelf', grain=CLOCK_COMPONENT)
+  CS%id_clock_pass = cpu_clock_id(' Ice shelf halo updates', grain=CLOCK_ROUTINE)
+
+  call cpu_clock_begin(CS%id_clock_pass)
   call pass_var(ISS%area_shelf_h, G%domain)
   call pass_var(ISS%h_shelf, G%domain)
   call pass_var(ISS%mass_shelf, G%domain)
   call pass_var(ISS%hmask, G%domain)
   call pass_var(G%bathyT, G%domain)
-  call cpu_clock_end(id_clock_pass)
+  call cpu_clock_end(CS%id_clock_pass)
+
 
   do j=jsd,jed ; do i=isd,ied
     if (ISS%area_shelf_h(i,j) > G%areaT(i,j)) then
       call MOM_error(WARNING,"Initialize_ice_shelf: area_shelf_h exceeds G%areaT.")
       ISS%area_shelf_h(i,j) = G%areaT(i,j)
     endif
   enddo ; enddo
-  if (present(fluxes)) then ; do j=jsd,jed ; do i=isd,ied
-    if (G%areaT(i,j) > 0.0) fluxes%frac_shelf_h(i,j) = ISS%area_shelf_h(i,j) / G%areaT(i,j)
+  if (present(fluxes_in)) then ; do j=jsd,jed ; do i=isd,ied
+    if (G%areaT(i,j)>0.) fluxes%frac_shelf_h(i,j) = ISS%area_shelf_h(i,j) / G%areaT(i,j)
   enddo ; enddo ; endif
 
   if (CS%debug) then
     call hchksum(fluxes%frac_shelf_h, "IS init: frac_shelf_h", G%HI, haloshift=0)
+    call hchksum(ISS%area_shelf_h, "IS init: area_shelf_h", G%HI, haloshift=0)
```

Comment:
I believe that the two `hchksum()` calls for `ISS%area_shelf_h` and `G%areaT` need dimensional rescaling arguments.